packages/components/src/filter-picker/index.js

```diff
@@ -143,7 +167,35 @@ class FilterPicker extends Component {
 		config.staticParams.forEach( ( param ) => {
 			update[ param ] = query[ param ];
 		} );
-		updateQueryString( update, path, persistedQuery );
+
+		// Remove all of this filter's params not associated witth the update while
+		// leaving any other params from any other filter an extension may have added.
+		this.getAllFilterParams().forEach( ( param ) => {
+			if ( ! update[ param ] ) {
+				// Explicitly give value of undefined so it can be removed from the query.
+				update[ param ] = undefined;
+			}
+		} );
+
+		const filterHasAdvancedFilters = !! config.filters.find(
+			( filter ) => filter.value === 'advanced'
+		);
+
+		// If the filter is associated with advancedFilters, remove advancedFilter parameters
+		// when navigating away from advancedFilters.
+		if ( filterHasAdvancedFilters && value !== 'advanced' ) {
```

Comment:
I think we should revert this back to the behavior in 005683c2ae89d134383e3238ac940ee2106349b7:

https://github.com/woocommerce/woocommerce-admin/blob/005683c2ae89d134383e3238ac940ee2106349b7/packages/components/src/filter-picker/index.js#L181

If we aren't fully decoupling this behavior, I think it should be as specific as possible so we don't potentially introduce some hard-to-diagnose bugs later on (like any 3PD filter that has a value of `advanced` resetting the advanced filters inadvertently).